src/protocol/tx-validity.md

```diff
@@ -205,13 +293,42 @@ Once the free balances are computed, the [script is executed](../fuel-vm/index.m
 1. The unspent free balance `unspentBalance` for each asset ID.
 1. The unspent gas `unspentGas` from the `$ggas` register.
 
-The fees incurred for a transaction are `ceiling(((size(tx) * GAS_PER_BYTE) + (tx.gasLimit - unspentGas) + sum(tx.inputs[i].predicateGasUsed)) * tx.gasPrice / GAS_PRICE_FACTOR)`.
-
-`size(tx)` includes the entire transaction serialized according to the transaction format, including witness data.
+`size(tx)` encompasses the entire transaction serialized according to the transaction format, including witness data.
 This ensures every byte of block space either on Fuel or corresponding DA layer can be accounted for.
 
 If the transaction as included in a block does not match this final transaction, the block is invalid.
 
+### Fees
+
+The cost incurred by a transaction can be calculated by:
+
+```py
+cost(tx) = gas_to_fee(gas_cost(tx) - unspentGas, tx.gasPrice)
+```
+
+where:
+
+- `gas_cost(tx)` is the final cost of the transaction in gas, including gas fees incurred from:
+  - The number of bytes comprising the transaction
+  - Processing inputs and outputs
+  - VM initialization
+  - Predicate and script execution
+- `unspentGas` is the amount gas left over after intrinsic fees and execution of the transaction, extracted from the `$ggas` register. Converting unspent gas to a fee describes how much "change" is left over from the user's payment; the block producer collects this unspent gas as reward.
+- `gas_to_fee` is a function that converts gas to a concrete fee based on a given gas price
+
+Users wishing to submit transactions can incentivize block producers to include their transactions by providing a higher reward in the form of more unspent gas. This is achieved by specifying a higher gas limit on the transaction.
+
+A naturally occurring result of a variable `gasLimit` is the concept of minimum and maximum fees. The minimum fee is thus the exact fee required to pay the fee balance, while the maximum fee is, then, the minimum fee plus any gas remaining after execution potentially paid to the producer as an incentive:
```

Comment:
I feel like maybe there should be an introduction/definition given for `gasLimit` before this statement.

Reply:
I agree that knowledge of what `gasLimit` is would be helpful for the reader; I wouldn't assert that an introduction to `gasLimit` belongs in the `tx-validity` specification since it is defined on the transaction independent of validity rules. Instead, I think I can provide a link to the transaction specification that mentions gas limit. For the reader's sake, it may behoove us to describe the components of a transaction, including `gasLimit` more in depth in that section.